### PR TITLE
Store peer address provided during peer add 

### PR DESCRIPTION
**Pull request by Madhu-1**
store peer addresses provided during peer add in ETCD.

if peer address already present in the store
update configuration on startup

this allows the users to use peer address
provided during peer add, for any other
operations(ex:- volume create,expand)

Fixes: #725 
Fixes: #758
Fixes: #724
Fixes: #721
Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by prashanthpai** [comment]
The commit message and PR description is ambiguous can be more clearer. You mention `peer details` multiple times, but this change is limited to peer addresses only.

**Comment by Madhu-1** [comment]
@prashanthpai  updated commit message and PR description

**Comment by prashanthpai** [comment]
@Madhu-1 Needs a rebase as https://github.com/gluster/glusterd2/pull/603 got merged

**Comment by Madhu-1** [comment]
@prashanthpai  rebased

File: glusterd2/commands/peers/addpeer.go [hunk 1]
```diff
@@ -99,10 +99,18 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		newpeer.Metadata[key] = value
 	}
 
+	//check if remotePeerAddress already present
+	found := utils.StringInSlice(remotePeerAddress, newpeer.PeerAddresses)
+	//if not found prepend the remotePeerAddress to peer details
+	if !found {
+		newpeer.PeerAddresses = append([]string{remotePeerAddress}, newpeer.PeerAddresses...)
+	}
```
Comment (aravindavk) [inline glusterd2/commands/peers/addpeer.go]: if found, we can move that address as first address in the list?

Comment (prashanthpai) [reply glusterd2/commands/peers/addpeer.go]: if not found, right ?

Comment (Madhu-1) [reply glusterd2/commands/peers/addpeer.go]: @aravindavk,  done

**Pull request merged**